modules/plugins/utility/motion/precognition/precognition.nix

```diff
@@ -0,0 +1,78 @@
+{lib, ...}: let
+  inherit (lib.options) mkEnableOption mkOption literalExpression;
+  inherit (lib.types) attrsOf listOf str bool int submodule;
+in {
+  options.vim.utility.motion.precognition = rec {
```

Comment:
Avoid `rec`. You can refer to the module's self with something like


```nix
let
	cfg = config.vim.utility.motiob.precognition;
in 
```

and use `cfg.<option>` instead.

Reply:
> ```nix
> 	cfg = config.vim.utility.motiob.precognition;
> ```


Question here - I used 'rec' in the option definitions because the syntax of `gutterHints` and `hints` are exactly the same and I believe they'll stay that way. If this is still a problem, I'll rewrite it.

Reply:
Top-level `rec` is both bad practice, and not allowed in nvf. It trips up LSPs, static analysis and reviewers.

You can refer to `hints` in `gutterHints` (which seems to be the only use case for recursion) with `cfg.setupOpts.hints`.

Reply:
I see! Thanks for the good knowledge, had no clue that using `rec` in the top level is not a good idea. Will get to fixing these up soon.

Reply:
Resolved in 39e27e17a2a63a5de7016a70bb2dcaf1180c1195